test/regression.js

```diff
@@ -160,3 +160,63 @@ test('regression: start replicating before feeds are loaded', function (t) {
   }
 })
 
+test('regression: announce new feed on existing connections', function(t) {
+  t.plan(19);
+  var m1 = multifeed(hypercore, ram, { valueEncoding: 'json' })
+  var m2 = multifeed(hypercore, ram, { valueEncoding: 'json' })
+  var m3 = multifeed(hypercore, ram, { valueEncoding: 'json' })
+
+  setup(m1, "First", function() {
+    setup(m2, "Second", function() {
+      setup(m3, "Third", function() {
+        var feedsReplicated = 0;
+        var r1 = null, r2 = null; // forward declare replication streams.
+
+        m1.on('feed', function(feed, name) {
+          feed.get(0, function(err, entry) {
+            t.error(err)
+            feedsReplicated++
+            switch(feedsReplicated) {
+              case 1: // First we should see M2's writer
+                t.equals(entry, "Second", "m2's writer should have been replicated")
```

Comment:
Bonus points for checking if `feed.key` is what you're expecting on each value of `feedsReplicated`!

Reply:
Good idea, let me add that.